src/clr-angular/data/datagrid/datagrid.spec.ts

```diff
@@ -44,13 +44,13 @@ export default function(): void {
 
             it("allows to manually resize the datagrid", function() {
```

Comment:
Missing some tests, particularly around projection based on display modes and perhaps some around the modification to row changes.

Reply:
Will add tests for display mode detection and projection for display and calculate modes.